File: src/main/java/pairtrix/domain/TeamSetup.java
package pairtrix.domain;

import java.time.LocalDate;
import java.util.List;

public class TeamSetup implements Comparable<TeamSetup>{

    private List<Pairing> pairings;
    private LocalDate dateReported;

    public TeamSetup(List<Pairing> pairings, LocalDate dateReported) {
        this.pairings = pairings;
        this.dateReported = dateReported;
    }

    public List<Pairing> getPairings() {
        return pairings;
    }

    @Override
    public int compareTo(TeamSetup teamSetup) {
        return teamSetup.getDateReported().compareTo(dateReported);
    }

    @Override
    public String toString() {
        return "TeamSetup{" +
                "pairings=" + pairings +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        TeamSetup teamSetup = (TeamSetup) o;

        return !(pairings != null ? !pairings.equals(teamSetup.pairings) : teamSetup.pairings != null);

    }

    @Override
    public int hashCode() {
        return pairings != null ? pairings.hashCode() : 0;
    }

    public LocalDate getDateReported() {
        return dateReported;
    }
}

File: pairtrix_use_case/src/main/java/pairtrix/domain/TeamSetup.kt
package pairtrix.domain

import java.time.LocalDate

data class TeamSetup(val pairings: Set<Pairing>, val dateReported: LocalDate) : Comparable<TeamSetup>, Set<Pairing> by pairings {

    override fun compareTo(other: TeamSetup): Int = other.dateReported.compareTo(dateReported)

    fun compairing(otherSetup: Set<Pairing>): Boolean {
        return intersect(otherSetup).isEmpty()
    }

}
